### ISSUE TITLE
Insert mode vertical bar cursor not working for nvim running within nvim

### ISSUE THREAD

**Issue by dstein64** [neovim/neovim#17173]
### Neovim version (nvim -v)

v0.6.1

### Vim (not Nvim) behaves the same?

Not applicable (related info in Actual behavior)

### Operating system/version

macOS 12.1

### Terminal name/version

Terminal.app 2.12

### $TERM environment variable

xterm-256color

### Installation

appimage

### How to reproduce the issue

Load Neovim from within Neovim.

```sh
$ nvim --clean
:terminal nvim --clean
```

Set `guicursor` to show a vertical bar in insert mode.

```vim
set guicursor=n-v-c:block,i-ci-ve:ver25
```

Enter insert mode.

```
i
```

### Expected behavior

Cursor would be a vertical bar.

### Actual behavior

Cursor is a block.

On Vim it's possible to set a vertical bar cursor for Vim running within Vim with:

```sh
$ vim -u DEFAULTS
:terminal vim -u DEFAULTS
```

```vim
let &t_SI = "\e[6 q"
let &t_EI = "\e[2 q"
```

```
i
```

**Comment by zeertzjq** [comment]
#3681 hasn't been fixed yet.

**Comment by zeertzjq** [comment]
Fixed by #31562
